### ISSUE TITLE
Breaking at the end of a case shouldn't be an error, even if unreachable

### ISSUE THREAD

**Issue by Gaelan** [microsoft/TypeScript#46479]
# Bug Report

### 🔎 Search Terms

switch, unreachable, break

### 🕗 Version & Regression Information

- This changed between versions 3.6.3 and 3.7.5

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.5.0-dev.20211022#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXxAA8sMAKASgC55UQA3EGAbgCgWV1s94ALKVYBBAAxHDlKIx1AORRp8AD7xpAI2nl4Abxbxd8AM4B3EmB4SxG7Xvhgo+hLOmUd1vWDz6cQgHQQcAc1IAIj4BCCxUf3goIPJWVz0VOCgAa3i3Owc1ZwTdd1RPH0YYHBhglXgsfXg0fWQAB3rSjBBgWPSEohIKDtckkFTegF8WIaA)

### 💻 Code

```ts
declare function exit(): never;

function handleFoo(foo: 'a' | 'b') {
    switch(foo) {
    case 'a':
        console.log("handling a");
        break;
    case 'b':
        console.error("b is unsupported");
        exit();
        break;
    }
}
```

### 🙁 Actual behavior

The `break` after `exit()` is flagged as unreachable code. This is technically true, but I'm not sure it's a good idea to remove it: many bugs have resulted from missing `break` statements, and people may want to make sure all cases end in a break, regardless of any functions that are called (either as a formalized code-style guideline, or just that little "this looks wrong" voice in your head).

### 🙂 Expected behavior

A `break` at the end of a switch statement shouldn't never be treated as unreachable code.

* * * 

I'm not quite sure if this qualifies as a bug report or a feature request; sorry if I made the wrong call.
